src/contracts/test/GPv2SettlementTestInterface.sol

```diff
@@ -18,4 +20,44 @@ contract GPv2SettlementTestInterface is GPv2Settlement {
     function allowanceManagerTest() external view returns (address) {
         return address(allowanceManager);
     }
+
+    function processTradesTest(
+        IERC20[] calldata tokens,
+        uint256[] calldata clearingPrices,
+        bytes calldata encodedTrades
+    )
+        external
+        view
+        returns (
+            GPv2AllowanceManager.Transfer[] memory inTransfers,
+            GPv2AllowanceManager.Transfer[] memory outTransfers
+        )
+    {
+        (inTransfers, outTransfers) = processTrades(
+            tokens,
+            clearingPrices,
+            encodedTrades
+        );
+    }
+
+    function processTradeMemoryTest() external pure returns (uint256 mem) {
+        GPv2Encoding.Trade memory trade;
+        GPv2AllowanceManager.Transfer memory inTransfer;
+        GPv2AllowanceManager.Transfer memory outTransfer;
+
+        // NOTE: Solidity stores the free memory pointer at address 0x40. Read
+        // it before and after calling `processOrder` to ensure that there are
+        // no memory allocations.
+        // solhint-disable-next-line no-inline-assembly
```

Comment:
We could disable this warning for the entire contract, at least for the test interfaces.

Reply:
I thought about it, and the main reason I decided to keep the lint (even if it annoying) is its a good way to enforce an explanatory comment everytime assembly is used.

> at least for the test interfaces

I think that is a good idea.

Reply:
I'm going to do this separatly as I think having a `function freeMemoryPointer()` is a nice utility to have that we can use for testing and in the actual contracts (instead of just having a `mload(0x40)` everywhere.